src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/MethodBuilder.cs

```diff
@@ -0,0 +1,422 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using Datadog.Trace.ClrProfiler.Emit;
+using Sigil;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
```

Comment:
I don't think this file should be in the `Emit` folder, not `Caching`. And the namespace should match the folder structure.
```suggestion
namespace Datadog.Trace.ClrProfiler.Integrations.Emit
```